job-application-tracker-api/src/users/users.service.ts

```diff
@@ -1,65 +1,101 @@
-import { Injectable } from '@nestjs/common';
+import { BadRequestException, Injectable } from '@nestjs/common';
 import { CreateUserDto } from './dto/create-user.dto';
 import { UpdateUserDto } from './dto/update-user.dto';
 import { NotFoundException } from '@nestjs/common';
+import { LoginDto } from './dto/login.dto';
+import { compare, hash } from 'bcrypt';
+import { User } from './models/user.model';
+import { JwtService } from '@nestjs/jwt';
+import { LoginResponse } from './dto/login-response.dto';
+import { JwtPayLoad } from 'src/shared/jwt-payload';
+import { Role } from 'src/shared/role';
+import { UserDto } from './dto/user.dto';
 
 @Injectable()
 export class UsersService {
-    private users = [
+    constructor(private readonly jwtService: JwtService,){}
+    
+    private users: User[] = [
         {
             "id": 1,
             "userName": "idoHashamen",
             "firstName": "ido",
             "lastName": "Rose",
             "email": "ido98@gmail.com",
-            "password": "123456"
-
+            "passwordHash": "$2a$10$oY3pF8WCLCsBMYFpSeTM.uld80NhHAnX797aasFRZbL1s8hxDMbxS",
+            "roles":[Role.user]
         },
         {
             "id": 2,
             "userName": "tomervak",
             "firstName": "tomer",
             "lastName": "vaknin",
             "email": "tomervak98@gmail.com",
-            "password": "123456"
+            "passwordHash": "$2a$10$oY3pF8WCLCsBMYFpSeTM.uld80NhHAnX797aasFRZbL1s8hxDMbxS",
+            "roles":[Role.user]
         },
         {
             "id": 3,
             "userName": "donfil",
             "firstName": "don",
             "lastName": "fil",
             "email": "bonfil98@gmail.com",
-            "password": "123456"
+            "passwordHash": "$2a$10$oY3pF8WCLCsBMYFpSeTM.uld80NhHAnX797aasFRZbL1s8hxDMbxS",
+            "roles":[Role.admin,Role.user]
         }
-    ]
+    ];
 
+    //TODO: replace with a mapper
+    getUserResponse(user: User) :UserDto {
```

Comment:
Create a file under users folder named ```user-mapper.ts```. Write a class called ```UserMapper``` and add a static function called ```toUserDto(user: User)``` and move this logic there.